### ISSUE TITLE
[bug] pyinstaller.py fails for conan2

### ISSUE THREAD

**Issue by gjasny** [conan-io/conan#12916]
### Environment details

* Operating System+version: macOS 13.1
* Compiler+version: Xcode 14
* Conan version: 2.0.0-beta8
* Python version: 3.9.6 (from macOS 13)

### Steps to reproduce

We use the following to create a standalone conan 1.57. It fails with conan 2.0.0-beta8 and current develop2 head (d7a53771573a85baeccd4909afd0d47a0c3fe82a):

```
/usr/bin/python3 -m venv ~/tmp/conan2-venv
source ~/tmp/conan2-venv/bin/activate

pip install -r conans/requirements.txt
pip install -r conans/requirements_server.txt
pip install -r conans/requirements_dev.txt

python pyinstaller.py
```

### Logs

```
16993 INFO: Building EXE from EXE-00.toc completed successfully.
16995 INFO: checking COLLECT
16995 INFO: Building COLLECT because COLLECT-00.toc is non existent
16996 INFO: Building COLLECT COLLECT-00.toc
18288 INFO: Building COLLECT COLLECT-00.toc completed successfully.
Initialized file: '/Users/gregorj/.conan2/settings.yml'
Initialized file: '/Users/gregorj/.conan2/extensions/plugins/compatibility/compatibility.py'
Initialized file: '/Users/gregorj/.conan2/extensions/plugins/compatibility/cppstd_compat.py'
Initialized file: '/Users/gregorj/.conan2/extensions/plugins/profile.py'
Traceback (most recent call last):
  File "conan/cli/cli.py", line 146, in run
    command_argument = args[0][0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "conans/conan.py", line 11, in <module>
    run()
  File "conans/conan.py", line 7, in run
    main(sys.argv[1:])
  File "conan/cli/cli.py", line 237, in main
    exit_error = cli.run(args)
  File "conan/cli/cli.py", line 148, in run
    self._output_help_cli()
  File "conan/cli/cli.py", line 110, in _output_help_cli
    max_len = max((len(c) for c in self._commands)) + 1
ValueError: max() arg is an empty sequence
[78656] Failed to execute script 'conan' due to unhandled exception!
Traceback (most recent call last):
  File "/Users/gregorj/src/conan/pyinstaller.py", line 135, in <module>
    output_folder = pyinstall(src_folder)
  File "/Users/gregorj/src/conan/pyinstaller.py", line 118, in pyinstall
    _run_bin(pyinstaller_path)
  File "/Users/gregorj/src/conan/pyinstaller.py", line 25, in _run_bin
    raise Exception("Binary not working")
Exception: Binary not working
```

Is `pyinstaller.py` still supported with Conan version 2? The same error happens on Ubuntu 22.04 and Windows.

Thanks,
Gregor

**Comment by memsharded** [comment]
Hi @gjasny 


Thanks for reporting this.
We haven't updated yet ``pyinstaller`` to work for 2.0, but the plan is to update it for the final 2.0 release (hopefully next month), so I am assigning this ticket to the milestone. Isn't it possible to use the ``pip`` installed beta for testing? I understand that the ``pyinstaller`` app is mostly for distributing to all devs or CI agents that don't have python easily installed, but for testing hopefully is enough?

**Comment by tim-goto** [comment]
FYI, the `pyinstaller.py` script runs successfully with python >= 3.10.9

**Comment by memsharded** [comment]
Proposing a fix in https://github.com/conan-io/conan/pull/13111, the problem is that pyinstaller was not bundling the new commands, which are loaded dynamically

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/13111, will be in next beta-10 (or 2.0 GA)

**Comment by gjasny** [comment]
@memsharded Could you please re-open? It still fails as before.

**Comment by memsharded** [comment]
It keeps working fine here, I have just tried, I cannot reproduce.
Did you update to latest ``develop2`` branch? Still exactly the same error? Maybe trying to clean your pyinstaller generated temp files?

**Comment by gjasny** [comment]
I saw the issue in our CI builds (which should start completely clean) across Linux, macOS, and Windows.

I can reproduce with the following Dockerfile:
```
FROM python:3.11

WORKDIR /testcase
RUN git clone --depth 1 -b develop2 https://github.com/conan-io/conan.git .

RUN pip install -r conans/requirements.txt
RUN pip install -r conans/requirements_server.txt
RUN pip install -r conans/requirements_dev.txt

RUN python pyinstaller.py
```

**Comment by gjasny** [comment]
It's still the same error:
```
Traceback (most recent call last):
  File "conan/cli/cli.py", line 146, in run
    command_argument = args[0][0]
                       ~~~~~~~^^^
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "conan/cli/cli.py", line 266, in main
    cli.run(args)
  File "conan/cli/cli.py", line 148, in run
    self._output_help_cli()
  File "conan/cli/cli.py", line 111, in _output_help_cli
    max_len = max((len(c) for c in self._commands)) + 1
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: max() arg is an empty sequence
```

**Comment by memsharded** [comment]
This is very weird. I have just tried with a couple of docker images, and it seems to work fine here.
Then I tried your code and failed. I have been trying a few things until I made this work:
```
FROM ubuntu:latest

RUN apt update
RUN apt install python3 git pip -y

WORKDIR /testcase
RUN git clone https://github.com/conan-io/conan.git .
RUN cd conan
RUN git checkout develop2

RUN pip install -e .

RUN python3 pyinstaller.py
```

Apparently the necessary thing to make it work is to define ``RUN pip install -e .``. because that puts itself in the necessary python paths, so the pyinstaller works fine, otherwise it doesn't find and bundle properly the dynamic commands.
